Mathlib/Data/Fintype/Card.lean

```diff
@@ -1030,6 +1030,12 @@ instance : Infinite ℕ :=
 instance Int.infinite : Infinite ℤ :=
   Infinite.of_injective Int.ofNat fun _ _ => Int.ofNat.inj
 
+instance String.infinite : Infinite String :=
+  Infinite.of_surjective String.length <| by
+    intro n
+    exists (String.replicate n 'x')
+    exact String.length_replicate n 'x'
```

Comment:
I think the proof is even shorter if you move it down a few lines and use `String.data` instead

Reply:
I've now changed the proof so it uses the injectivity of `String.mk`. I suspect that said injectivity could be proven more concisely, do you have any suggestions in that regard?